src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -325,11 +335,31 @@ def _forward(self, model_inputs):
                 # Send stride to `postprocess`.
                 # it needs to be handled there where
                 # the pieces are to be concatenated.
-                ratio = 1 / self.model.config.inputs_to_logits_ratio
+                if input_to_features_ratio is None:
+                    ratio = self.model.config.inputs_to_logits_ratio
+                else:
+                    new_ratio = inputs.shape[1] / logits.shape[1]
+                    if isinstance(input_to_features_ratio, list):
+                        f = input_to_features_ratio[0]
+                        batch_size = len(input_to_features_ratio)
+                        if any([abs(f - r) / f > 1e-2 for r in input_to_features_ratio]):
+                            # This shouldn't trigger since ratios should be consistent
+                            # through the pipeline, however since everything is calculated
+                            # a check shouldn't hurt
+                            raise Exception("The ratios are not similar enough")
+                        ratio = input_to_features_ratio[0] * new_ratio
+                        outratio = [ratio] * batch_size
+                    else:
+                        ratio = input_to_features_ratio * new_ratio
+                        outratio = ratio
+                    out["ratio"] = outratio
```

Comment:
This code is overly complex, but given the end results seems odd, I didn't refactor it yet into something cleaner.

Reply:
It works as expected, meaning I'm cutting where I should.
But the *results* however (the reconstructed string) is lacking part of the stride.

That's because we cannot attribute produced letters to specific logits, so the cutting logic is unable to know how to reconstruct the string. I don't think it's a property of the mode.

Reply:
That's fine for me honestly - M-CTCT is not used a lot at all